x-pack/legacy/plugins/ml/public/contexts/kibana/kibana_context.ts

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+
+import { IndexPattern } from 'ui/index_patterns';
+
+export interface KibanaContextValue {
+  combinedQuery?: any;
```

Comment:
i'm unsure about this `combinedQuery` being in here. Firstly because I believe it's only used for creation pages, e.g. job wizards, data frame wizard.
But mainly because it's not a standard kibana thing. it was originally created for the wizards to differentiate between the query you get from saved search and the result of combining that query with the filters and index pattern (also from the saved search) using kibana's `buildEsQuery`.
Renaming it to `query` might be to general and cause confusion. I'm also not suggesting removing it now, as it's being used. but i think it's worth discussing as there may be other things like this which could go into their own context.
Also I think its type should be `Query`
from
`import { Query } from 'src/legacy/core_plugins/data/public';`

Reply:
It depends on `... = Private(SearchItemsProvider)` so it depends on being injected via angular (hence the previous change to `AngularContext` which we reverted). Even if the name of the context is `KibanaContext` now, its purpose is to pass on angularjs dependencies. I'm open to other suggestions but I'd like to address this in a follow up.

Reply:
OK cool, I'm happy for it to stay in. We might need to revisit it later if most consumers of `KibanaContext` don't require it.